src/sparse/node.rs

```diff
@@ -412,30 +425,30 @@ where
     StorageType: StorageInspect<NodesTable>,
     StorageType::Error: fmt::Debug,
 {
-    pub fn left_child(&self) -> Option<Self> {
+    pub fn left_child(&self) -> Result<Option<Self>, DeserializeError> {
```

Comment:
`Result<Option<>>` is ugly, is there any possible route around this?

Reply:
I think `Result<Option<...>, ...>` is, in itself, a fair pattern. It represents two things independently: 1. Was the operation successful? And 2. Was the requested record found? We use this pattern in things like our `Storage` traits, which these child getters will often wrap - getting a child node can be implemented as a database read, where there is the possibility the record is not found or a database read error occurs. 

However, in this specific case, I am considering how this interface is actually used, since that informs our the decision if this is the right approach. In practice, getting the child can be fallible, but if the child is not found, that could be considered an error. For example, traversing a Merkle tree requires iterating from the root to a leaf, retrieving the nodes' children along the way. If a child node is not found, it necessarily means an invalid tree or corrupted database, both of which represent error cases.

I am currently exploring these options in a WIP PR. The related issue is https://github.com/FuelLabs/fuel-merkle/issues/132.